ports/stm/supervisor/port.c

```diff
@@ -454,3 +454,14 @@ void _init(void)
 
 }
 
+void HAL_Delay(uint32_t delay_ms) {
+    mp_hal_delay_ms(delay_ms);
+}
+
+#if 0
+uint32_t HAL_GetTick() {
+    uint8_t subticks;
+    uint32_t result = (uint32_t)port_get_raw_ticks(&subticks);
+    return (result * 1000) + (subticks * 1000 / 32);
```

Comment:
I'm not totally sure what to do here. In Micropython's low power implementation, HAL_GetTick isn't overridden at all - they use the system version, which depends on SysTick. HAL_Delay dynamically selects whether to run a SysTick based delay or a busy loop depending on whether interrupts are enabled or not. 

I don't know how anything in the HAL could even be working without HAL_GetTick since it's used more or less constantly for timeouts, which implies to me that the Systick is currently turned on. I'll try comparing the raw ticks reading to what the default HAL_GetTick returns and see if I can figure out why this isn't working.